zebra-state/src/service/non_finalized_state.rs

```diff
@@ -182,6 +182,7 @@ impl NonFinalizedState {
         // Chain::cmp uses the partial cumulative work, and the hash of the tip block.
         // Neither of these fields has interior mutability.
         // (And when the tip block is dropped for a chain, the chain is also dropped.)
+        #[allow(clippy::mutable_key_type)]
```

Comment:
This was a [false positive](https://rust-lang.github.io/rust-clippy/master/index.html#/mutable_key_type) and should remain a false positive unless there are unexpected changes to how `Chain` is mutated or its `Ord` impl. 

We could refactor this method to use `pop_last()` instead, I'm happy to do that in this PR, open an issue, or leave a TODO if we want to do that.